### ISSUE TITLE
unknown signature key algorithm: 22

### ISSUE THREAD

**Issue by nlmarco** [bcgit/bc-java#371]
Since last month, reading my key-ring fails with this exception:

```
Caused by: java.io.IOException: unknown signature key algorithm: 22
        at org.bouncycastle.bcpg.SignaturePacket.<init>(Unknown Source) ~[bcpg-jdk15on-1.56.jar:1.56.0.0]
        at org.bouncycastle.bcpg.BCPGInputStream.readPacket(Unknown Source) ~[bcpg-jdk15on-1.56.jar:1.56.0.0]
        at org.bouncycastle.openpgp.PGPKeyRing.readSignaturesAndTrust(Unknown Source) ~[bcpg-jdk15on-1.56.jar:1.56.0.0]
        at org.bouncycastle.openpgp.PGPKeyRing.readUserIDs(Unknown Source) ~[bcpg-jdk15on-1.56.jar:1.56.0.0]
        at org.bouncycastle.openpgp.PGPPublicKeyRing.<init>(Unknown Source) ~[bcpg-jdk15on-1.56.jar:1.56.0.0]
        at org.bouncycastle.openpgp.PGPObjectFactory.nextObject(Unknown Source) ~[bcpg-jdk15on-1.56.jar:1.56.0.0]
        at org.bouncycastle.openpgp.PGPPublicKeyRingCollection.<init>(Unknown Source) ~[bcpg-jdk15on-1.56.jar:1.56.0.0]
        at org.subshare.core.pgp.gnupg.BcWithLocalGnuPgPgp.load(BcWithLocalGnuPgPgp.java:763) ~[org.subshare.core-0.10.4-SNAPSHOT.jar:na]
        ... 13 common frames omitted
```
It seems, Enigmail automatically updated my public key-ring -- and repeats doing so, whenever I revert it. Using [subshare](https://github.com/subshare) together with Thunderbird/Enigmail/GnuPG, with both sharing the same key-ring thus does not work, anymore.

There already is an issue for bc-csharp: https://github.com/bcgit/bc-csharp/issues/73

But since these are separate code-bases and my problem https://github.com/subshare/subshare/issues/61 is caused by bc-java (I don't use C#), I assume I should open an issue here. I hope, this is correct.

The problem is very likely the same: The algorithm with the ID 22 is not supported, neither in the Java- nor in the C#-version of BC.

**Comment by nlmarco** [comment]
I'd like to add that the same problem still happens with BC 1.59, too. The stack-trace above was taken, before I upgraded to the newest BC.

**Comment by bcgit** [comment]
Okay, I think this is actually fixed in the latest beta of 1.60 - b10. Note we're not trying to support EdDSA at this point, just get through without parse errors.

See https://www.bouncycastle.org/betas - let us know how you go.

**Comment by bcgit** [comment]
Closing this as it seems the 1.61 release would have dealt with it.
